File: 383-ransom-note/ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int alpha[26] = {0};

        for(char a : magazine) alpha[a - 'a']++;

        for(char a:ransomNote) if(alpha[a-'a']-- <=0) return false;

        return true;
    }
};

File: 383-ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine):
            return False
        ans = {}

        for a in ransomNote:
            ans[a] = ans.get(a,0) + 1

        for a in ans.keys():
            if magazine.count(a) < ans[a]:
                return False
        return True